File: pkg/doctor/testdata-packs/collection-method/positive.rb
require 'stripe'

# Subscription create — top-level billing
Stripe::Subscription.create(
  customer: 'cus_x',
  billing: 'charge_automatically',
)

# Subscription update — top-level billing
Stripe::Subscription.update(
  'sub_x',
  billing: 'send_invoice',
)

# SubscriptionSchedule create — billing nested under default_settings AND under a phase
Stripe::SubscriptionSchedule.create(
  customer: 'cus_x',
  default_settings: {
    billing: 'charge_automatically',
  },
  phases: [
    {
      billing: 'charge_automatically',
      items: [{price: 'price_x'}],
    },
  ],
)

File: pkg/doctor/testdata-packs/prorate/positive.js
const stripe = require('stripe')('sk_test_x');

await stripe.subscriptions.update('sub_1MnGA42eZvKYlo2ClPYPojkP', {
  prorate: false,
});

await stripe.subscriptionItems.create({
  subscription: 'sub_1MnGA42eZvKYlo2ClPYPojkP',
  price: 'price_1MoBy5LkdIwHu7ixZhnattbh',
  prorate: true,
});
